packages/wrangler/src/api/startDevWorker/ConfigController.ts

```diff
@@ -352,33 +374,46 @@ export class ConfigController extends Controller<ConfigControllerEventMap> {
 		const signal = this.#abortController.signal;
 		this.latestInput = input;
 
-		const fileConfig = readConfig(input.config, {
-			env: input.env,
-			"dispatch-namespace": undefined,
-			"legacy-env": !input.legacy?.enableServiceEnvironments ?? true,
-			remote: input.dev?.remote,
-			upstreamProtocol:
-				input.dev?.origin?.secure === undefined
-					? undefined
-					: input.dev?.origin?.secure
-						? "https"
-						: "http",
-			localProtocol:
-				input.dev?.server?.secure === undefined
-					? undefined
-					: input.dev?.server?.secure
-						? "https"
-						: "http",
-		});
-		void this.#ensureWatchingConfig(fileConfig.configPath);
-
-		const resolvedConfig = await resolveConfig(fileConfig, input);
-		if (signal.aborted) {
-			return;
+		try {
```

Comment:
basically for the same reason as in https://github.com/cloudflare/workers-sdk/pull/6574/files#r1733250847. But I would like some sanity check on whether this gets executed only in watch mode. If not, we might need to rethink the logic here, in case throwing and exiting the dev process is expected to happen in some use cases